### PR TITLE
Custom max deferral timing

### PR DESCRIPTION
**Pull request by mpanighetti**
- added option for custom `$MAX_DEFERRAL_TIME` and `$SKIP_DEFERRAL` settings defined by plist attributes, or if undefined, reverts to script default (allows for managing deferral periods via configuration profile rather than making the change in the script and repackaging)
- standardized casing (`UpperCamelCase` for CFPreferences, `ALL_CAPS_WITH_UNDERSCORES` for variables)
- split README text in definition lists into multiple paragraphs

### CODE REVIEW

**Comment by christopherdietrich** [comment]
Are you sure this works with a config profile?
I don't think deploying a profile will actually update the value in the plist,
maybe use `python -c "from Foundation import CFPreferencesCopyAppValue; print CFPreferencesCopyAppValue('MAX_DEFERRAL_TIME', '$PLIST')"` instead of `defaults read...`?

**Comment by mpanighetti** [comment]
It’s worked in my testing, though there might be a delay before the system starts picking up the custom value; killing `cfprefsd` can force the system to recognize the new setting sooner for testing purposes.

File: payload/Library/Scripts/install_or_defer.sh [hunk 1]
```diff
@@ -362,6 +368,17 @@ if [[ -z "$LOGO" ]] || [[ ! -f "$LOGO" ]]; then
     LOGO="/System/Library/CoreServices/Software Update.app/Contents/Resources/SoftwareUpdate.icns"
 fi
 
+# Validate max deferral time. To set this to a custom value, make a
+# configuration profile enforcing the MAX_DEFERRAL_TIME attribute in $PLIST to
+# a positive integer of your choice.
+MAX_DEFERRAL_TIME_PLIST=$(defaults read "$PLIST" MAX_DEFERRAL_TIME 2>/dev/null)
```
[the remark this answers is not in the document]
Comment (homebysix) [reply payload/Library/Scripts/install_or_defer.sh]: Fixed in 79ee310.

File: payload/Library/Scripts/install_or_defer.sh [hunk 2]
```diff
@@ -94,9 +94,15 @@ HARD_RESTART_DELAY=$(( 60 * 5 )) # (300 = 5 minutes)
 # Created by: perreal (http://stackoverflow.com/users/390913/perreal)
 convert_seconds () {
 
-    ((h=${1}/3600))
-    ((m=(${1}%3600)/60))
-    ((s=${1}%60))
+    if [[ $1 -eq 0 ]]; then
```
[the remark this answers is not in the document]
Comment (mpanighetti) [reply payload/Library/Scripts/install_or_defer.sh]: Yeah, I ran into issues where it was treating the zero as null and breaking the conversion.

**Pull request merged**